lib/src/lib.nr

```diff
@@ -218,12 +206,30 @@ impl<PROOF_LEN, MAX_VALUE_LEN> TrieProof<20, PROOF_LEN, MAX_VALUE_LEN>
 /// # Arguments
 /// * `dest` - Destination array
 /// * `src` - Source array
-/// * `start_pos` - Starting position in source array 
-fn memcpy<N, M>(dest: &mut [u8; N], src: [u8; M], start_pos: u64)
+/// * `offset` - Offset in source array 
+fn memcpy<N, M>(dest: &mut [u8; N], src: [u8; M], offset: u64)
+{
+    for i in 0..N
+    {
+        (*dest)[i] = src[offset + i];
+    }
+}
+
+/// Verifies that specified number of elements from a subarray match elements from the array
+/// at specified offset.
+///
+/// # Arguments
+/// * `subarray` - Array to be verified
+/// * `array` - Array to be verified against
+/// * `len` - Number of elements to be verified
+/// * `offset` - Offset in `array`
+fn assert_arrays_equal<N, M>(subarray: [u8; N], array: [u8; M], len: u64, offset: u64)
```

Comment:
Could you rename this to something like `assert_subarray` to indicate the asymmetry in the arguments?